bluetooth: tester: bap: Add support for BAP/USR tests

mkasenberg opened the pull request:
Main changes:
- Extend Codec capabilities of the server.
- At Codec Config operation initiated autonomously by server, all ASEs with lower IDs than the requested one have to be initiated first due to API limitation.
- In some BAP/USR (server role) test cases the PTS sends the MMI 202 to configure the Sink ASE to streaming state, in some others it does not. For now, let's autonomously send Receiver Start Ready at ASE Enabled.

mkasenberg committed:
bluetooth: tester: bap: Add support for BAP/USR tests

Extend Codec capabilities of the server.

At Codec Config operation initiated autonomously by server, all ASEs
with lower IDs than the requested one have to be initated first due
to API limitation.

In some BAP/USR (server role) test cases the PTS sends the MMI 202
to configure the Sink ASE to streaming state, in some others it does not.
For now, let's autonomously send Receiver Start Ready at ASE Enabled.

Signed-off-by: Magdalena Kasenberg <magdalena.kasenberg@codecoup.pl>

sjanc commented on tests/bluetooth/tester/src/btp_bap.c:
```diff
@@ -524,8 +526,25 @@ static void stream_qos_set(struct bt_bap_stream *stream)
+
+	err = bt_bap_ep_get_info(stream->ep, &info);
+	if (err) {
+		LOG_ERR("Could not get EP info for stream %p", stream);
```
just return here,  if we cannot get ep info we won't be able to properly fill in event data anyway (as info will be left uninitialized).

or maybe send complete event but with failed status?

(same in stream_strated)

Thalley commented on tests/bluetooth/tester/src/btp_bap.c:
```diff
@@ -524,8 +526,27 @@ static void stream_qos_set(struct bt_bap_stream *stream)
+		return;
+	}
+
+	(void)bt_conn_get_info(stream->conn, &conn_info);
```
Any reason why we are ignoring the return value of `bt_conn_get_info` but not of `bt_bap_ep_get_info`? 

IMO makes sense to treat these 2 the same

Thalley commented on tests/bluetooth/tester/src/btp_bap.c:
```diff
@@ -1050,7 +1068,11 @@ static uint8_t bap_send(const void *cmd, uint16_t cmd_len,
 
 	stream = stream_find(audio_conn, cp->ase_id);
-	if (stream == NULL || stream->stream.ep->dir != BT_AUDIO_DIR_SINK) {
+	if (stream == NULL ||
+	    (conn_info.role == BT_HCI_ROLE_CENTRAL &&
+	     stream->stream.ep->dir != BT_AUDIO_DIR_SINK) ||
+	    (conn_info.role == BT_HCI_ROLE_PERIPHERAL &&
+	     stream->stream.ep->dir != BT_AUDIO_DIR_SOURCE)) {
```
The `bt_bap_ep_info` now contains a `can_send` field to make it easier to verify if you can TX on a given stream/ep

Thalley commented on tests/bluetooth/tester/src/btp_bap.c:
```diff
@@ -1290,6 +1320,81 @@ static bool codec_config_store(struct bt_data *data, void *user_data)
+	struct bt_bap_ep *ep;
+	struct audio_stream *stream;
+
+	stream = stream_find(audio_conn, ase_id);
+
+	if (stream == NULL) {
```
```suggestion
	stream = stream_find(audio_conn, ase_id);
	if (stream == NULL) {
```

Thalley commented on tests/bluetooth/tester/src/btp_bap.c:
```diff
@@ -1290,6 +1320,81 @@ static bool codec_config_store(struct bt_data *data, void *user_data)
+			return -EINVAL;
+		}
+
+		ep = end_point_find(audio_conn, ase_id);
+
+		if (ep == NULL) {
```
```suggestion
		ep = end_point_find(audio_conn, ase_id);
		if (ep == NULL) {
```

Thalley commented on tests/bluetooth/tester/src/btp_bap.c:
```diff
@@ -1290,6 +1320,81 @@ static bool codec_config_store(struct bt_data *data, void *user_data)
+	struct audio_stream *stream;
+	int err = 0;
+
+	stream = stream_find(audio_conn, ase_id);
+
+	if (stream == NULL) {
```
```suggestion
	stream = stream_find(audio_conn, ase_id);
	if (stream == NULL) {
```

Thalley reviewed (dismissed):
LGTM with a few suggestions

fabiobaltieri commented:
@mkasenberg please rebase.

carlescufi commented:
@mkasenberg please rebase

mkasenberg committed:
Bluetooth: Audio: Fix missing setting of data and meta len

A recent refactor of bt_audio_codec_cap to flat arrays missed setting
data_len and meta_len fields.

Signed-off-by: Magdalena Kasenberg <magdalena.kasenberg@codecoup.pl>
